zipline/src/engineTest/kotlin/app/cash/zipline/SourceMapParserTest.kt

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (C) 2021 Square, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package app.cash.zipline
+
+import app.cash.zipline.internal.SourceMapParser
+import app.cash.zipline.internal.readVarint
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+import okio.Buffer
+import okio.FileSystem
+import okio.Path.Companion.toPath
+
+class SourceMapParserTest {
+  @Test fun simpleSourceMap() {
+    val sourceMapJson = """
+      {
+        "version": 3,
+        "file": "kotlin-js-demo.js",
+        "sources": [
+          "../../../../../src/main/kotlin/main.kt"
+        ],
+        "sourcesContent": [
+          null
+        ],
+        "names": [],
+        "mappings": ";;;;;;;;;;;;EAAA,gB;IACE,YAAY,c;IACZ,OAAQ,KAAI,KAAJ,C;EACV,C;;;;;;"
```

Comment:
Does this contain a segment with an optional name? I'm missing a few golden `Segment` instances here that are asserted again to ensure the logic is fully covered.

Reply:
Looks like kotlin-js doesn't populate `names` when generating the map, so we shouldn't be expecting to see any names in real-world code. Would probably still be a good idea to run this code against a map generated by something else, I can look into that in a followup. Meanwhile made the test a bit more descriptive and added a case for a line that doesn't map to anything in the source: 903d35a